registry/options.go

```diff
@@ -18,7 +18,9 @@ type Options struct {
 }
 
 type RegisterOptions struct {
-	TTL time.Duration
+	TCPCheck bool
```

Comment:
We will not accept TCPCheck as an Option value here. Put the TCPCheck in the Context as with option Consul specific options.